libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/AILTNPublisher.java

```diff
@@ -129,6 +138,28 @@ public boolean publishLogLines(JSONArray logLines) {
         return false;
     }
 
+    private RequestBody setContentLength(final RequestBody requestBody) throws IOException {
+        final Buffer buffer = new Buffer();
+        requestBody.writeTo(buffer);
+        return new RequestBody() {
+
+            @Override
+            public MediaType contentType() {
+                return requestBody.contentType();
+            }
+
+            @Override
+            public long contentLength() {
+                return buffer.size();
```

Comment:
We need to buffer the whole thing first to get the exact size to set `Content-Length` header. Some servers are sensitive to this and return `411` if it's missing, such as our community endpoints. It works fine against our org endpoints though.